deepspeed/inference/engine.py

```diff
@@ -73,8 +73,8 @@ def __init__(self, model, config):
         if hasattr(self.module, "config"):
             TransformerPolicy.hf_model_config = self.module.config
 
-        if config.dtype == torch.half and not get_accelerator().is_fp16_supported():
-            raise ValueError("Type fp16 is not supported.")
+        if config.dtype not in get_accelerator().is_fp16_supported():
```

Comment:
@nelyahu - I believe this is failing because it should check 

```
if config.dtype not in get_accelerator().supported_dtypes():
```

Reply:
@nelyahu - figured I could just make this change now and wait for you to confirm if this is acceptable.